### PR TITLE
Try to improve initial example of a reducer comp

### PR DESCRIPTION
**Pull request by slorber**
Because it was not clear to me that the initial example was not a valid code snippet. I expected it to compile after pasting it in my editor.

### CODE REVIEW

**Comment by slorber** [comment]
I tried to make it more clear, but as I'm new to Reason I don't really know what I'm doing ;)

File: docs/state-actions-reducer.md [hunk 1]
```diff
@@ -8,12 +8,19 @@ ReasonReact stateful components are like ReactJS stateful components, except wit
...
+
+type state = int
+
 let component = ReasonReact.reducerComponent("Greeting");
 
 let make = (~name, _children) => {
   ...component,
-  initialState: () => 0, /* here, state is an `int` */
+  initialState: () => 42,  /* here, state is an `int` */
+  reducer: (action,state) => { ... },
```
Comment (yawaramin) [inline docs/state-actions-reducer.md]: This won't compile either after you paste it into your project, right? :-)

Comment (slorber) [reply docs/state-actions-reducer.md]: sure it does not but at least it's explicit it won't compile :)

**Comment by sanderhahn** [comment]
With some help on Discord i got to this working Greeting example:

```reasonml
type state = int;

type action =
  | Increment;

let component = ReasonReact.reducerComponent("Greeting");

let text = ReasonReact.stringToElement;

let make = (~name, _children) => {
  ...component,
  initialState: () => 0,
  reducer: (action, state) =>
    switch action {
    | Increment => ReasonReact.Update(state + 1)
  },
  render: (self) => {
    let greeting =
      "Hello " ++ name ++ ". You've clicked the button " ++ string_of_int(self.state) ++ " time(s)!";
    <div>
      {text(greeting)}
      <button onClick={_event => self.send(Increment)}>{text("Click Me")}</button>
    </div>
  }
};
```

**Comment by yawaramin** [comment]
Sébastien, looks like @chenglou has updated the example in https://github.com/reasonml/reason-react/commit/7eed3579ebfc7402eebe331b740b0a8bed4e44b5

**Comment by slorber** [comment]
great I'm closing this then

**Comment by chenglou** [comment]
Thanks for the contribution! Yeah we've received similar feedback from others so we decided to just sync this snippet with the one in the bsb boilerplate, etc.

**Pull request closed without merging**